### PR TITLE
vsphere: Initial CAA implementation

### PR DESCRIPTION
**Pull request by caavery**
This pull request contains the patch which is the initial peer pod cloud-api-adapter for vsphere vcenter.

### CODE REVIEW

**Comment by yoheiueda** [comment]
Thank you very much for contributing this patch.

Could you follow the commit message format described here?
https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#main-patch

File: pkg/adaptor/hypervisor/vsphere/vsphere.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		// vm.Destroy will fail if the VM is still powered on.
+		_ = task.Wait(ctx)
+	}
+
+	task, err = vm.Destroy(ctx)
+	if err != nil {
+		return err
+	}
+
+	task.Wait(ctx)
```
Comment (bpradipt) [inline pkg/adaptor/hypervisor/vsphere/vsphere.go:272]: @caavery is this needed ?

Comment (caavery) [reply pkg/adaptor/hypervisor/vsphere/vsphere.go:272]: In all the vsphere go code I've looked at it is common practice to wait for the destroy to complete

Comment (caavery) [reply pkg/adaptor/hypervisor/vsphere/vsphere.go:272]: So would this fix it? 
_ = task.Wait(ctx)

Comment (bpradipt) [reply pkg/adaptor/hypervisor/vsphere/vsphere.go:272]: Yeah you can use the above

**Review by yoheiueda — APPROVED** [review]
LGTM

**Pull request merged**